workspace/src/app/views/shared/RuleEditor/model/RuleEditorModel.tsx

```diff
@@ -1422,6 +1670,47 @@ export const RuleEditorModel = ({ children }: RuleEditorModelProps) => {
         }, 1);
     };
 
+    const addStickyNoteToCanvas = (stickyNote: string, color: string, reactFlowWrapper) => {
```

Comment:
There are several things that are not clear to me:
- This is called `addStickyNoteToCanvas`, but it also seems to update existing sticky notes. Mabye there should be 2 functions instead?
- Why is `currentStickyContent` used, e.g. why not use an optional node ID as parameter?
- Both functions ` changeStickyNodeStyle` and `changeStickyNoteContent` are both part of the context, but only used internally in the rule editor model.